crates/flox-rust-sdk/src/providers/git.rs

```diff
@@ -30,7 +29,6 @@ pub struct BranchInfo {
 
 // simple git provider for the tasks we need to provide in
 // flox
-#[async_trait(?Send)]
 pub trait GitProvider: Send + Sized + std::fmt::Debug {
```

Comment:
If we aren't using `async` you can remove the `Send + Sized` bounds, those are generally only required for `async` code because `tokio` uses a work-stealing threadpool and thus Futures need to be thread safe. If you remove these bounds we would be able to have nicer APIs in the future if we have time to refactor.

Reply:
`discover`, `init`, and `clone` return `Self`, so don't those still need `Sized`?

Reply:
I think we're just going to scrap the trait entirely though

Reply:
At the very least remove `Send` so we don't _need_ to use owned types.

Reply:
Should be a pretty quick edit

Reply:
I'd rather also remove the trait altogether. Having half the methods on the trait and anther half on the impl is weird especially since we only really use one git impl after all.
Its out of scope for this issue though!